src/Lumi/Styles/Button.purs

```diff
@@ -0,0 +1,241 @@
+module Lumi.Styles.Button where
+
+import Prelude
+
+import Color (cssStringHSLA, darken)
+import Data.Foldable (fold)
+import Data.Monoid (guard)
+import Lumi.Components.Size (Size(..))
+import Lumi.Components.ZIndex (ziButtonGroup)
+import Lumi.Styles.Box (FlexAlign(..), align, interactive, justify, row)
+import Lumi.Styles.Theme (LumiTheme)
+import React.Basic.Emotion (Style, color, css, int, merge, selector, str)
+
+data ButtonKind
+  = Primary
+  | Secondary
+
+-- | Outline
+-- | Link
+data ButtonState
+  = Enabled
+  | Disabled
+  | Loading
+
+button :: LumiTheme -> ButtonKind -> ButtonState -> Size -> Style
+button theme buttonKind buttonState size =
```

Comment:
Could you explain please the reason for why we'd provide this style as a value separate from the `button` component?

Reply:
Component composition is more parent-child, while style composition behaves like mixins/overrides. I demo'd that "all parent-child composition" approach last week and it seemed like we didn't like it as much, so this leans the other way where everything that can be defined as pure style is, and at some point components come in to add more involved behavior. In this case, the button styles define all the visual states and rules, while the component focuses on connecting the onClick and other html element details like type and disabled (the inability to click, not the way it looks).

Reply:
At the moment our "anchor tags that look like buttons" are part of the button component, but if they were their own component they'd be able to use these button styles to make some other element look like a button.

Reply:
Hm yes, that makes sense. Thank you for explaining :)